operators/src/source/gdal_source/mod.rs

```diff
@@ -673,7 +644,21 @@ where
         (tile_x_size, tile_y_size),       // requested raster size
         None,                             // sampling mode
     )?;
-    Grid::new(tile_grid, buffer.data, no_data_value).map_err(Into::into)
+    let data = Grid::new(tile_grid, buffer.data)?;
+
+    let masked_grid = MaskedGrid::from(data).map_or_mask_elements(|e| {
+        // TODO: test if parallel map is better.
```

Comment:
yeah I guess we should test that. Also we should look into storing our dataset as masked raster bands to avoid doing this conversion over and over and over and over again.

Also: didn't you say Gdal allows somehow accessing a band as a masked band?

Reply:
1. yes. there are a lot of places where we could use the parallel map. we should do that in an new task.
2. gdal can do that. we have to implement it first.

Reply:
can you add some docs to the NoDataValueGrid? At first glance once could think that it's a grid full of no data.

Reply:
[47ea273](https://github.com/geo-engine/geoengine/pull/561/commits/47ea27352a728b4d7f22efc52bbc45d594f3f0cc)